components/file_upload/file_upload.jsx

```diff
@@ -565,15 +565,16 @@ class FileUpload extends PureComponent {
         this.setState({menuOpen: false});
     }
 
-    simulateInputClick = () => {
+    simulateInputClick = (e) => {
+        e.preventDefault();
+        e.stopPropagation();
```

Comment:
i may be wrong but i remember looking into this at one point and attempting something similar but it still opened up the file picker twice for me. i.e. was seeing https://mattermost.atlassian.net/browse/MM-20939 behaviour. 

@sudheerDev can you please confirm if you're not able to reproduce the issue mentioned in MM-20939 with this change?

Reply:
using chrome/ff

Reply:
yes, i am positive it does not cause a regression with the ticket you mentioned.

Reply:
Ahh, it does not cause that issue but it has not fixed the issue of picker not opening without double click. It worked lcaoclly for me. Gotto check this again

Reply:
👍

Reply:
@ali-farooq0 The PR works. I don't know what i was testing earlier. It fixes the ticket i linked and it does not cause a regression with the ticket you linked.